### PR TITLE
Preliminary cleanup of mapview_common

### PR DESCRIPTION
**Pull request by lmoureaux**
This PR contains a collection of small commits that improve the quality of the `mapview_common` code that handles drawing the map and updating it when needed. The commits are mostly independent from each other and I'm happy to take out controversial ones and discuss them in another PR. No behavior change is expected.

I strongly suggest to review this commit by commit. See individual commit messages for more context.

### CODE REVIEW

**Comment by lmoureaux** [comment]
The Apple clang failure should be fixed, others look like glitches...

File: client/mapview_common.cpp [hunk 1]
```diff
@@ -1650,7 +1648,6 @@ void decrease_unit_hp_smooth(struct unit *punit0, int hp0,
     }
 
     unqueue_mapview_updates(true);
-    gui_flush();
```
Comment (pranavtifr) [inline client/mapview_common.cpp:1653]: Just curious.. Dont we need some sort of repaint here after the units are updated on mapcanvas ?

File: client/mapctrl_common.cpp [hunk 2]
```diff
@@ -374,41 +374,3 @@ void update_line(int canvas_x, int canvas_y)
...
-
-/**
-   We sort according to the following logic:
-
-   - Transported units should immediately follow their transporter (note that
-     transporting may be recursive).
-   - Otherwise we sort by ID (which is what the list is originally sorted
- by).
- */
-static int unit_list_compare(const void *a, const void *b)
```
Comment (pranavtifr) [inline client/mapctrl_common.cpp:385]: Hmm, I remember something breaking when I removed this sometime ago :smile: But, somehow it works ontop of everything else you did.

**Comment by pranavtifr** [comment]
> The Apple clang failure should be fixed, others look like glitches...

Looks like the opposite to me, windows ones are linker errors which we need to fix apple one seems like a vcpkg issue.

**Comment by pranavtifr** [comment]
Anyways, the code refactor looks ok. We can merge if the builds dont break.

File: client/mapview_common.cpp (hunk 1 above)
Comment (lmoureaux) [reply client/mapview_common.cpp:1653]: `unqueue_mapview_updates(true)` already calls `flush_dirty()`

**Comment by jwrober** [comment]
Maybe we need to rebase this to master branch and let the CI run again?

**Comment by lmoureaux** [comment]
> Maybe we need to rebase this to master branch and let the CI run again?

I'm waiting for Windows Update to complete then I can investigate. Probably tomorrow.

**Comment by jwrober** [comment]
> > Maybe we need to rebase this to master branch and let the CI run again?
> 
> I'm waiting for Windows Update to complete then I can investigate. Probably tomorrow.

It fails on my local Visual Studio too, i'm trying to find the error message.

**Comment by lmoureaux** [comment]
Builds for me without SDL and without updating my MSYS install.
```
$ cmake --build build -j5
-- Git HEAD Commit Hash: (hex) ea9fe7 and (dec) 15376359
-- Last tag commit hash: 137068d83d196a4e9c8391365886996a9581e54d
-- Latest Git Tag: v3.0-beta.3
SDL2 not found
SDL2_mixer not found
-- Found Lua: C:/msys64/mingw64/lib/liblua.dll.a (found suitable version "5.4.3", minimum required is "5.3")
-- Could NOT find ToLuaProgram (missing: TOLUA_PROGRAM)
-- Found Lua: C:/msys64/mingw64/lib/liblua.dll.a (found version "5.4.3")
-- Using the Windows native API for stack unwinding. This is the preferred option.
-- Using the Windows native API to retrieve stack symbols. You'll need to convert DWARF debug information to PDB.
-- Could NOT find Sphinx (missing: SPHINX_EXECUTABLE)
-- Including CPack
-- Configuring done
-- Generating done
-- Build files have been written to: C:/msys64/home/Louis/freeciv21/build
```

**Comment by lmoureaux** [comment]
After installing SDL, it fails with:
```
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/11.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:/msys64/mingw64/lib/libqtmain.a(qtmain_win.o):C:\M\mingw-w64-qt5-pkgs\src\MINGW64\qtbase\src\winmain/qtmain_win.cpp:97: undefined reference to `qMain(int, char**)'
collect2.exe: error: ld returned 1 exit status
```
--> hints at an interaction between Qt and SDL

**Comment by lmoureaux** [comment]
f87d3d588aab352a4f3e9253f25b9c4d553159e4 is the first bad commit

**Comment by lmoureaux** [comment]
Windows build fixed in 2edf4ecb8466e4eaa0dbb25e5e9b2ed498b455a3

**Pull request merged**